upgrade_tests/drop_compact_storage_upgrade_test.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+import logging
+
+from dtest import Tester
+
+since = pytest.mark.since
+logger = logging.getLogger(__name__)
+
+VERSION_30 = 'github:apache/cassandra-3.0'
+VERSION_311 = 'github:apache/cassandra-3.11'
+
+
+@since('3.0', max_version='3.11')
+class TestDropCompactStorage(Tester):
+    """
+    Test to verify that dropping compact storage is not possible prior running `nodetool upgradesstables`.
+    @CASSANDRA-16063
+    """
+
+    def test_drop_compact_storage(self):
+        cluster = self.cluster
+        cluster.populate(2)
+        node1, node2 = cluster.nodelist()
+        cluster.set_install_dir(version="2.1.14")
+        cluster.start(wait_for_binary_proto=True)
+
+        session = self.patient_cql_connection(node1)
+        self._insert_data(session)
+
+        logging.debug("Upgrading to current version")
+        for node in [node1, node2]:
+            node.drain()
+            node.watch_log_for("DRAINED")
+            node.stop(wait_other_notice=False)
+
+            self.set_node_to_current_version(node)
+            node.start(wait_for_binary_proto=True)
+
+        session = self.patient_cql_connection(node1)
+        thrown = False
+        exception = None
+        try:
+            session.execute("ALTER TABLE drop_compact_storage_test.test DROP COMPACT STORAGE")
+        except Exception as e:
+            exception = e
+            thrown = True
+
+        assert thrown, "No exception has been thrown"
+        assert "Cannot DROP COMPACT STORAGE until all SSTables are upgraded, please run `nodetool upgradesstables` first." in str(exception)
+
+        for node in [node1, node2]:
+            node.nodetool("upgradesstables")
+
+        thrown = False
+        exception = None
+        try:
+            session.execute("ALTER TABLE drop_compact_storage_test.test DROP COMPACT STORAGE")
+        except Exception as e:
+            exception = e
+            thrown = True
+
+        assert not thrown, "Exception should not have been thrown: " + str(exception)
```

Comment:
Perhaps we could just use a simpler call to `session.execute`, without the try block and the variables:
```suggestion
        session.execute("ALTER TABLE drop_compact_storage_test.test DROP COMPACT STORAGE")
```